src/tests/ftest/util/data_mover_test_base.py

```diff
@@ -0,0 +1,398 @@
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+The Government's rights to use, modify, reproduce, release, perform, display,
+or disclose this software are subject to the terms of the Apache License as
+provided in Contract No. B609815.
+Any reproduction of computer software, computer software documentation, or
+portions thereof marked with this legend must also reproduce the markings.
+"""
+from command_utils_base import CommandFailure
+from daos_utils import DaosCommand
+from test_utils_pool import TestPool
+from test_utils_container import TestContainer
+from ior_test_base import IorTestBase
+from data_mover_utils import DataMover
+from os.path import join
+import uuid
+
+class DataMoverTestBase(IorTestBase):
+    # pylint: disable=too-many-ancestors
+    """Base DataMover test class.
+
+    Sample Use Case:
+        set_ior_location_and_run("DAOS_UUID", "/testFile, pool1, cont1,
+                                 flags="-w -K")
+        set_src_location("DAOS_UUID", "/testFile", pool1, cont1)
+        set_dst_location("POSIX", "/some/posix/path/testFile")
+        set_ior_location_and_run("POSIX", "/some/posix/path/testFile",
+                                 flags="-r -R")
+    :avocado: recursive
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Initialize a DataMoverTestBase object."""
+        super(DataMoverTestBase, self).__init__(*args, **kwargs)
+        self.dm_cmd = None
+        self.processes = None
+        self.pool = []
+        self.containers = []
+        self.uuids = []
+        self._gen_daos_path_v = 0
+
+    def setUp(self):
+        """Set up each test case."""
+        # Start the servers and agents
+        super(DataMoverTestBase, self).setUp()
+
+        self.dfuse_hosts = self.agent_managers[0].hosts
+
+        # Get the parameters for DataMover
+        self.dm_cmd = DataMover(self.hostlist_clients)
+        self.dm_cmd.get_params(self)
+        self.processes = self.params.get("np", '/run/datamover/processes/*')
+
+    # The valid parameter types for setting param locations.
+    PARAM_TYPES = ("POSIX", "DAOS_UUID", "DAOS_UNS")
+
+    def validate_param_type(self, param_type):
+        """Validates the param_type, converts to upper-case,
+           handles shorthand types.
+
+        Args:
+            param_type (str): The param_type to be validated.
+
+        Returns:
+            (str) A valid param_type
+        """
+        s = str(param_type).upper()
```

Comment:
Minor: can we change 's' to some word like 'type'?

Reply:
Yes, thank you